### ISSUE TITLE
mixup not applied to weak labels

### ISSUE THREAD

**Issue by fschmid56** [DCASE-REPO/DESED_task#93]
Hi,

if I'm not mistaken, mixup is applied to weakly labeled data in the following line (in file [sed_trainer_pretrained.py](https://github.com/DCASE-REPO/DESED_task/blob/1cb8c84f848240b77bc9ba6894c3d5f3b76283a7/recipes/dcase2024_task4_baseline/local/sed_trainer_pretrained.py#L339)):

`features, embeddings, labels = self.apply_mixup(features, embeddings, labels, indx_strong, indx_weak)`

However, the variable _labels_weak_ is already assigned before and not modified after mixup. Is it intended to not mix the labels for weakly labeled clips, or am I overlooking something?

**Comment by popcornell** [comment]
No you are correct thank you for noticing it. 
I have fixed it and will push the changes along with a new training script with Optuna that participants might find useful for hyperparameter tuning. 

BTW from preliminary experiments it strangely seems it is not affecting much the performance

**Comment by fschmid56** [comment]
Thanks! I noticed the same; it doesn't have a big performance impact, which is surprising to me.

**Comment by popcornell** [comment]
This has been solved. Pushed the new code
